csrc/root_domain_map.cpp

```diff
@@ -747,19 +747,20 @@ void ComputeAtRootDomainMapBuilder::initializeBcastMap(
     return;
   }
 
-  // This initialization should be only used for: 1) fusion output tensors, 2)
-  // outputs of multi-consumer expressions that are not fusion outputs, 3) view
-  // outputs as broadcasts can be merged with non-broadcast domains, resulting
-  // in non-broadcast rfactor domains, and 4) squeeze inputs as broadcasts can
-  // be removed by squeeze.
-  TORCH_INTERNAL_ASSERT(
-      tv->isFusionOutput() || ir_utils::isSqueezeInput(tv) ||
-          tv->definition()->outputs().size() > 1 ||
-          tv->isDefinitionType<ViewOp>(),
-      "Invalid tensor to initialize bcast map t",
-      tv->name(),
-      " in tensor ",
-      tv->toString());
+  // This initialization of the entry for the broadcast ID should only
+  // happen when the broadcast has no further consumer ID, including
+  // resolved non-broadcast IDs. An equivalent condition is that the
+  // pairwise map has no mapping for the broadcast.
+  for (auto consumer : ir_utils::consumerTvsOf(tv)) {
+    const auto p2c =
+        PairwiseRootDomainMap(tv, consumer)
+            .mapProducerToConsumer(tv->domain(), consumer->domain());
+    // Unfortunately, const_cast is required as our const model is
+    // broken.
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-const-cast)
+    TORCH_INTERNAL_ASSERT(p2c.find(const_cast<IterDomain*>(id)) == p2c.end());
+  }
```

Comment:
This is another bug fix. It's just a sanity check of the overall analysis, but this check should be more thorough and should not give false positives.